website/payments/api/v2/admin/views.py

```diff
@@ -0,0 +1,144 @@
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        return Response(
+            PaymentSerializer(
+                serializer.instance, context=self.get_serializer_context()
+            ).data,
+            status=status.HTTP_201_CREATED,
+        )
+
+
+class PaymentDetailView(AdminRetrieveAPIView, AdminDestroyAPIView):
+    queryset = Payment.objects.all()
+    serializer_class = PaymentSerializer
+    permission_classes = [IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["payments:admin"]
+
+
+class PayableBaseView:
+    required_scopes = ["payments:admin"]
+
+    def get_permissions(self):
+        return [IsAuthenticatedOrTokenHasScope()]
+
+    def get_payable(self):
+        app_label = self.kwargs["app_label"]
+        model_name = self.kwargs["model_name"]
+        payable_pk = self.kwargs["payable_pk"]
+
+        try:
+            payable_model = apps.get_model(app_label=app_label, model_name=model_name)
+            payable = payables.get_payable(
+                get_object_or_404(payable_model, pk=payable_pk)
+            )
+        except (LookupError, NotRegistered) as e:
+            raise serializers.ValidationError(
+                {api_settings.NON_FIELD_ERRORS_KEY: [_("Payable model not found")]}
+            ) from e
+
+        if not payable.can_manage_payment(self.request.member):
+            raise PermissionDenied(
+                detail=_("You do not have permission to perform this action.")
+            )
+
+        return payable
+
+
+class PayableDetailView(
+    PayableBaseView, AdminRetrieveAPIView, AdminUpdateAPIView, AdminDestroyAPIView
+):
+    """View that allows you to manipulate the payment for the payable. Permissions of this view are based on the payable."""
+
+    def get_serializer_class(self, *args, **kwargs):
+        if self.request.method.lower() == "patch":
+            return PayableCreateSerializer
+        return PayableSerializer
+
+    def get(self, request, *args, **kwargs):
+        serializer = self.get_serializer(self.get_payable())
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def delete(self, request, *args, **kwargs):
+        payable = self.get_payable()
+
+        if not payable.model.payment:
+            raise Http404
+
+        try:
+            services.delete_payment(
+                payable.model, request.member,
+            )
+            payable.model.save()
+        except PaymentError as e:
+            raise PermissionDenied(detail=str(e))
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def patch(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        payable = self.get_payable()
+
+        try:
+            services.create_payment(
```

Comment:
```suggestion
            payable.model.payment = services.create_payment(
```

Reply:
That is not required, the code in services already does that.